### PR TITLE
Add Slack subdomain

### PR DESCRIPTION
**Pull request by firejake308**
Added test for Slack subdomain. This PR replaces #317, but if preferred, I could also make a PR to @loleg's repo and then merge them into this repo via #317.

### CODE REVIEW

**Comment by loleg** [comment]
Thanks so much for this! I am closing my PR in favor of this one.

File: flask_dance/contrib/slack.py [hunk 1]
```diff
@@ -56,19 +57,27 @@ def make_slack_blueprint(
...
+        hostname = "slack.com"
     slack_bp = SlackBlueprint(
         "slack",
         __name__,
         client_id=client_id,
         client_secret=client_secret,
         scope=scope,
         base_url="https://slack.com/api/",
-        authorization_url="https://slack.com/oauth/authorize",
+        authorization_url="https://{hostname}/oauth/authorize".format(
```
Comment (daenney) [inline flask_dance/contrib/slack.py:78]: You can do this as `authorization_url = "https://slack.com/oauth/authorize" if subdomain is None else "https://{subdomain}.slack.com/oauth/authorize".format(subdomain=subdomain)`.

Comment (firejake308) [reply flask_dance/contrib/slack.py]: Makes sense. Applied changes in latest commits.

**Review by daenney — APPROVED** [review]

**Comment by loleg** [comment]
@singingwolfboy anything I could do to help get this into a [release tag](https://github.com/singingwolfboy/flask-dance/compare/v3.2.0...main) ?

**Comment by singingwolfboy** [comment]
@loleg I just released v3.3.0, with this change included. Thanks for the reminder!

**Comment by loleg** [comment]
Thank you so much

**Pull request merged**